src/remote/manager.go

```diff
@@ -106,47 +124,79 @@ func (u *UnitOfWork) Invoke() error {
 	return u.errors.ErrorOrNil()
 }
 
-func (u *UnitOfWork) createOrUpdate(object model.ObjectState, changedFields []string) error {
+func (u *UnitOfWork) createOrUpdate(objectState model.ObjectState, changedFields []string) error {
+	// Get object local state.
+	// Remote state is used for marking object as deleted (then local state is not set)
+	var object model.Object
+	if objectState.HasLocalState() {
+		object = objectState.LocalState()
+	} else if objectState.HasRemoteState() {
+		object = objectState.RemoteState()
+	} else {
+		panic(fmt.Errorf(`local or remote state must be set`))
+	}
+
 	// Set changeDescription
 	switch v := object.(type) {
-	case *model.ConfigState:
-		v.Local.ChangeDescription = u.changeDescription
+	case *model.Config:
+		v.ChangeDescription = u.changeDescription
 		changedFields = append(changedFields, "changeDescription")
-	case *model.ConfigRowState:
-		v.Local.ChangeDescription = u.changeDescription
+	case *model.ConfigRow:
+		v.ChangeDescription = u.changeDescription
 		changedFields = append(changedFields, "changeDescription")
 	}
 
 	// Create or update
-	if !object.HasRemoteState() {
+	if !objectState.HasRemoteState() {
 		// Create
-		request, err := u.api.CreateRequest(object.LocalState())
-		if err != nil {
-			return err
-		}
+		return u.create(objectState, object)
+	}
+
+	// Update
+	return u.update(objectState, object, changedFields)
+}
+
+func (u *UnitOfWork) create(objectState model.ObjectState, object model.Object) error {
+	request, err := u.api.CreateRequest(object)
+	if err != nil {
+		return err
+	}
+	u.poolFor(object.Level()).
+		Request(request).
+		OnSuccess(func(response *client.Response) {
+			// Save new ID to manifest
+			objectState.SetRemoteState(object)
+
+			// If marking as deleted -> local state is not set
+			if objectState.HasLocalState() {
+				u.localManager.UpdatePaths(objectState, false)
+				if err := u.localManager.SaveModel(objectState.Manifest(), objectState.LocalState()); err != nil {
+					u.errors.Append(err)
+				}
+			}
+		}).
+		OnError(func(response *client.Response) {
+			if e, ok := response.Error().(*Error); ok {
+				if e.ErrCode == "configurationAlreadyExists" || e.ErrCode == "configurationRowAlreadyExists" {
```

Comment:
Priklad:
- na servery je config `[TO DELETE] abc`
- tj. pre CLI neexistuje
- lokalne mam stav, ked ten config este nebol zmazany, tj. `abc`.
- ak zavolam `push` tak sa CLI tool snazi konfig vytvorit, ... ale kedze uz existuje, tak dostaneme chybu.
- takze spravime nasledne `update` namiesto `create`.
- a config teda opat existuje, z nazvu sa odstranil prefix `[TO DELETE]`, ... tj. spravili sme `undelete`.
- `update` ak objekt existuje sa moze hodit, aj ak nastane nejaka data-race, ... alebo sa spusti viac CLI instancii naraz -> a stav sa zmeni medzi vytvorenim planu a spustenim.